### PR TITLE
Fix #2626: Trivial module accessors not elided in 2.12

### LINKED ISSUE #2626: Trivial module accessors not elided in 2.12
Consider:

``` scala
js.Dynamic.console.log(1)
```

in 2.12.0-RC1, this yields the following code:

``` javascript
$g.console.log(($m_sjs_js_Any$(), 1))
```

Pre 2.12, `js.Any` (needed for implicit conversion) was elided.

### PR DESCRIPTION
**Pull request by gzm0**

### CODE REVIEW

**Comment by gzm0** [comment]
I manually verified that module accessors are indeed elided in 2.12.

**Comment by gzm0** [comment]
Review by @sjrd

File: tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/optimizer/GenIncOptimizer.scala [hunk 1]
```diff
@@ -620,26 +620,39 @@ abstract class GenIncOptimizer private[optimizer] (semantics: Semantics,
...
         case ApplyStatic(ClassType(cls), methodName, List(This())) =>
           statics(cls).methods(methodName.name).originalDef.body.exists {
             case Skip() => true
             case _      => false
           }
+
+        // Mixin constructor, 2.12
+        case ApplyStatically(This(), ClassType(cls), methodName, Nil)
+            if !classes.contains(cls) =>
+
```
Comment (sjrd) [inline tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/optimizer/GenIncOptimizer.scala]: This empty line is weird. I suggest removing it.

**Comment by gzm0** [comment]
Updated. I have also:

- Fixed checksizes to point to 2.12.0-RC2
- Tightened checksizes

**Comment by sjrd** [comment]
The test failure doesn't seem spurious. It's doing something weird with checksizes.

**Comment by gzm0** [comment]
> It's doing something weird with checksizes.

I "fixed" checksizes to use 2.12-RC2 "like the rest". Seems "the rest" does not use 2.12-RC2 :P

Updated.

**Comment by sjrd** [comment]
```
Checksizes: Scala version: 2.12.0-RC1
Reversi preopt size = 593680 (expected 560000)
Reversi opt size = 148668 (expected 140000)
Reversi preopt gzip size = 77147 (expected 76500)
Reversi opt gzip size = 35537 (expected 35500)
```

**Comment by gzm0** [comment]
Updated.

**Pull request merged**